Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainActivity.java

```diff
@@ -355,4 +365,21 @@ public void onTabSelected(@NonNull TabLayout.Tab tab)
     showGames();
     GameFileCacheManager.startLoad(this);
   }
+
+  @Override
+  public void setTheme(int themeId)
+  {
+    super.setTheme(themeId);
+    this.themeId = themeId;
+  }
+
+  public int getThemeId()
+  {
+    return themeId;
+  }
+
+  private void checkTheme()
+  {
+    ThemeHelper.checkThemeSetting(this);
```

Comment:
Only calling this in MainActivity means that Dolphin's settings system will never be updated with new values from SharedPreferences if you don't pass through MainActivity. For instance, you could be using TvMainActivity, or you could be launching AppLinkActivity through a frontend app.

What do you think about moving the logic of updating Dolphin's settings system into DirectoryInitialization? We already have some code there for syncing Dolphin's settings system with SharedPreferences for Wii Remote settings. Then MainActivity would run `ThemeHelper.setCorrectTheme` after directory initialization finishes instead of `ThemeHelper.checkThemeSetting`.

Reply:
I put something together quick to move things into DirectoryInit. Let me know what you think.